src/rgw/rgw_op.cc

```diff
@@ -3749,7 +3922,10 @@ int RGWDeleteObj::handle_slo_manifest(bufferlist& bl)
 
 int RGWDeleteObj::verify_permission()
 {
-  if (!verify_bucket_permission(s, RGW_PERM_WRITE)) {
+  if (!verify_object_permission(s,
+				s->object.instance.empty() ?
```

Comment:
@adamemerson : Did you replace verify_bucket_permissions() with verify_object_permissions() by mistake? Replacing it back with the correct method call, makes TestAccountNoContainers of swift tests to pass.